### ISSUE TITLE
Lua equivalent to FlxG.sound.music.pause();?

### ISSUE THREAD

**Issue by greentophat** [ShadowMario/FNF-PsychEngine#6599]
### Describe your problem here.

I'm creating a custom pause in a lua script, starting with a few lines to hide almost every ui element, some to make a custom lua sprite and text visible, and ending with return Function_Stop; to prevent the vanilla one from executing.
So I made persistentUpdate and persistentDraw false, made paused true, and paused the vocals.
But then I ran into a snag.
pauseSound('music'); didn't work
neither did pauseSound(getPropertyFromClass('FlxG.sound', 'music'));
so, i'm hoping there's someone here who knows what i could do to pause the inst.

TL;DR, I'm trying to make a piece of lua code that functions identical to FlxG.sound.music.pause(); but can't figure out how

### Are you modding a build from source or with Lua?

Lua

### What is your build target?

Windows x64

**Comment by ShadowMario** [comment]
Buffer the time where it was paused, set the volume to 0 and make it go to time 0 until it's unpaused (then set the time back to where you paused it)

**Comment by greentophat** [comment]
just to be clear, would that look like

```
local buffer = 0 --storing the song position in this
function onPause()
	buffer = getSongPosition(); --buffer song position
	setPropertyFromClass('flixel.FlxG.sound.music', 'volume', 0);
	setProperty('vocals.volume', 0);
        return Function_Stop;
end
function onResume()
	           setPropertyFromClass('Conductor', 'songPosition', buffer);
	           setPropertyFromClass('flixel.FlxG.sound.music', 'volume', 1);
	           setProperty('vocals.volume', 1);
end
function onUpdatePost(elapsed)
        if getProperty('paused') then
	           setPropertyFromClass('Conductor', 'songPosition', 0);
                   end
end
```

this?

**Comment by greentophat** [comment]
i'd also like to note that my method of doing this appears to literally not work at all
it saves songposition to the buffer and according to the debug it's actually trying to set the songposition to 0
but when i UN-pause, it's revealed that the song's been going the entire time and doesn't at all set its position to the buffer value.
